etna/models/catboost.py

```diff
@@ -53,7 +57,21 @@ def fit(self, df: pd.DataFrame, regressors: List[str]) -> "_CatBoostAdapter":
         """
         features = df.drop(columns=["timestamp", "target"])
         target = df["target"]
-        self._categorical = features.select_dtypes(include=["category"]).columns.to_list()
+        columns_dtypes = features.dtypes
+        category_columns_dtypes = columns_dtypes[columns_dtypes == "category"]
+        self._categorical = category_columns_dtypes.index.tolist()
+
+        # select only columns with float categories
+        float_category_columns_dtypes_indices = [
+            idx
+            for idx, x in enumerate(category_columns_dtypes)
+            if issubclass(x.categories.dtype.type, (float, np.floating))
+        ]
+        float_category_columns_dtypes = category_columns_dtypes.iloc[float_category_columns_dtypes_indices]
+        float_category_columns = float_category_columns_dtypes.index
+        self._float_category_columns = float_category_columns
+        self._prepare_float_category_columns(features)
```

Comment:
You return df there. I think it's better to assign it df here.

Current side effects are not obvious

Reply:
Currently the df is changing inplace (we add new columns), so may be it is better to not return anything instead of making a copy.

Reply:
I see, but it should be more explicit

Reply:
I made this function to not return anything. I think it is more clear that it makes inplace transformation.